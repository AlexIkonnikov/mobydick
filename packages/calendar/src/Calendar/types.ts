import {MarkingProps} from 'react-native-calendars/src/calendar/day/marking';
import {CalendarProps} from 'react-native-calendars';
import {ReactElement} from 'react';

export interface IMarkedTypes {
  [key: string]: MarkingProps;
}

export interface IMarkedDates {
  dates: IMarkedTypes;
  fromDate: Date;
  toDate: Date;
}
export type colorElem = {
  color: string;
  textColor: string;
};

export type IColors = {
  colorPrime: colorElem;
  colorSoft: colorElem;
};

export interface IDateRange {
  dateStart: string;
  dateEnd: string;
}

export enum IButtonView {
  small = 'small',
  large = 'large',
}

export interface ICalendar extends CalendarProps {
  onDateRangeChange?: (dateRange: IDateRange) => void;
  bottomView?: ReactElement;
  defaultLocale?: string;
  isClear?: boolean;
  isShowToday?: boolean;
  localeConfig?: ILocaleConfig | undefined;
  isPeriod?: boolean;
}

export interface ILocaleConfig {
  monthNames: string[];
  monthNamesShort: string[];
  dayNames: string[];
  dayNamesShort: string[];
  today?: string;
}
