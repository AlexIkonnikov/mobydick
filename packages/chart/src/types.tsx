import type {ReactElement} from 'react';
import type {SharedValue} from 'react-native-reanimated';

export interface ICoordinates {
  x: number;
  y: number;
}

export interface IChart {
  name?: string;
  coordinates: ICoordinates[];
  colors?: string[];
}

export interface IDataset {
  [key: string]: IChart[];
}

export type IFormatter = (value: number) => string;

export interface IGeneratePeriodsWithPaths {
  dataset: IDataset;
  width: number;
  height: number;
}

export type IRenderSectionItem = (
  {
    period,
    state,
    transition,
  }: {
    period: string;
    state: ISharedGraphState;
    transition: IChartTransition;
  },
  index: number,
) => ReactElement;

export interface IGraphState {
  next: number;
  current: number;
}

export type ISharedGraphState = SharedValue<IGraphState>;
export type IChartTransition = SharedValue<number>;
