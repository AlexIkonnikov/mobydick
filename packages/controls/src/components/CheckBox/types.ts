import React from 'react';

import {IControlProps} from '../types';

interface ICheckBox {
  children: React.ReactNode;
}

export type TCheckBox = ICheckBox & Omit<IControlProps, 'text'>;
