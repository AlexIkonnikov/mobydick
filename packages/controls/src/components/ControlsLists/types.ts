import {FunctionComponentElement} from 'react';
import {PressableProps} from '@npm/mobydick-core';

import {IControlProps} from '../types';

export interface IControlsList {
  onChange(values: string[]): void;
  children: FunctionComponentElement<IControlProps & PressableProps>[];
  horizontal?: boolean;
  single?: boolean;
}
