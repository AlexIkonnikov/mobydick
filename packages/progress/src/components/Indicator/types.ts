import {SvgProps} from 'react-native-svg';
import {ViewStyle} from 'react-native';

interface IIndicatorProps {
  percent: number;
  indicatorViewStyles?: ViewStyle;
  indicatorStyles?: ViewStyle;
}

export type IndicatorProps = IIndicatorProps & SvgProps;
