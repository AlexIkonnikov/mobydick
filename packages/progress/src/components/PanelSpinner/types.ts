interface IPanelSpinnerProps {
  isLoading: boolean;
  duration?: number;
  isError?: boolean;
  onCancel?: () => void;
}
export type PanelSpinnerProps = IPanelSpinnerProps;
