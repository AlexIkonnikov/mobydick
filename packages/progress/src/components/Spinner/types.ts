import {SvgProps} from 'react-native-svg';

interface ISpinnerProps {
  speed?: number;
}
export type SpinnerProps = ISpinnerProps & SvgProps;
