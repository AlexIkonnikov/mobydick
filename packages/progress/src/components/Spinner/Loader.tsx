import {LoaderS, LoaderM, LoaderL, useTheme} from '@npm/mobydick-styles';
import React, {FC} from 'react';
import {Color} from 'react-native-svg';

import {ISizeSpinner, LoaderProps} from './types';

const Loader: FC<LoaderProps> = props => {
  const {size, fill, ...otherProps} = props;
  const {colors} = useTheme();

  const getSize = (size: ISizeSpinner, color: Color) => {
    switch (size) {
      case ISizeSpinner.S:
        return <LoaderS fill={color} {...otherProps} />;
      case ISizeSpinner.M:
        return <LoaderM fill={color} {...otherProps} />;
      case ISizeSpinner.L:
        return <LoaderL fill={color} {...otherProps} />;
    }
  };

  return getSize(size, fill || colors.ElementBase);
};

export default Loader;
