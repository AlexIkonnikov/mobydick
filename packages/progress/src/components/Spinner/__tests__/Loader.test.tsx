import React from 'react';
import {render} from '@testing-library/react-native';

import {ISizeSpinner} from '../types';
import Loader from '../Loader';

describe('@npm/mobydick-progress/Loader', () => {
  it('renders correctly S', () => {
    const {toJSON} = render(<Loader size={ISizeSpinner.S} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly M', () => {
    const {toJSON} = render(<Loader size={ISizeSpinner.M} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly L', () => {
    const {toJSON} = render(<Loader size={ISizeSpinner.L} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
