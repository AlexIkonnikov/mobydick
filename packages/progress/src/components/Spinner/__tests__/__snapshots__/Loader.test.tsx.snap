// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-progress/Loader renders correctly L 1`] = `
<SvgMock
  fill="#2B78EE"
  height="48"
  width="48"
/>
`;

exports[`@npm/mobydick-progress/Loader renders correctly M 1`] = `
<SvgMock
  fill="#2B78EE"
  height="32"
  width="32"
/>
`;

exports[`@npm/mobydick-progress/Loader renders correctly S 1`] = `
<SvgMock
  fill="#2B78EE"
  height="26"
  width="26"
/>
`;

exports[`@npm/mobydick-progress/Loader renders correctly without size 1`] = `
<SvgMock
  fill="#2B78EE"
  height="24"
  width="24"
/>
`;
